### PR TITLE
Add quakenet auth support

### PR DESCRIPTION
**Pull request by BiohZn**
Adds simple support for auth on quakenet

### CODE REVIEW

**Comment by JohnMaguire** [comment]
Related to #98 (different forms of auth).

I wonder if maybe we could just convert this into something more similar to #106? i.e. A plugin that registers commands to send on boot -- then AUTH could be sent, or "/msg Nickserv", or what-have you. This also allows for getting IRC server implementation details out of core.

May involve making sure plugins boot before the connection is fully created.

File: cardinal/bot.py [hunk 1]
```diff
@@ -99,9 +99,20 @@ def signedOn(self):
 
         # Set the currently connected network
         self.network = self.factory.network
-
+        self.network_alias = self.factory.network.split('.')[-2]
+
+        if self.network_alias == 'quakenet':
```
Comment (JohnMaguire) [inline cardinal/bot.py]: If we add a condition for quakenet into the core, I'd rather have it toggle-able by a config option, than by hostname (geographically-specific hostnames could break on this, e.g.).

I also think we can send AUTH directly to the server through `cardinal.send()`.

Comment (BiohZn) [inline cardinal/bot.py]: geographically-specified should not break this as I use the [-2]
using send though sounds reasonable :)

Comment (JohnMaguire) [inline cardinal/bot.py]: Ah, nice! I missed the `-` heh. :)

**Comment by JohnMaguire** [comment]
@BiohZn I like this better, since GameSurge recommends trying `AUTH` first. If you update the merg branch to `release/2.3`, I will merge this.

However, it may get reverted by the time I actually cut 2.3 -- it must go into 2.3 per semver (feature added), but I'm hoping to get the plugin order (deps) & communication done by 2.3. It's possible this means that auth drivers through plugins will be implemented in 2.3. If not though, this'll be a great stop-gap!

Thanks again. :) :+1:

**Comment by JohnMaguire** [comment]
Okay, I merged this into release/2.3 (4fa2cd7). I was curious however if you might be willing to implement a couple extra things (and if not, I'll get around to them eventually):

1. If `authname` is not set, default to `nick`.
2. Use the same name for `AUTH` as we would for `NickServ` (i.e. add the username param to `IDENTIFY`).

**Comment by BiohZn** [comment]
I can implement those too, no problem

**Pull request closed without merging**